File: frontend/src/services/storage.service.ts
import api from "@/lib/axios";

export interface StorageUsageResponse {
  usedBytes: number;
  limitBytes: number;
  remainingBytes: number;
  usedPercentage: number;
}

export const StorageService = {
  async getUsage(): Promise<StorageUsageResponse> {
    const response = await api.get<StorageUsageResponse>("/api/storage/usage");
    return response.data;
  },
};

export default StorageService;

File: src/main/java/com/skydrive/skydrive/service/StorageService.java
package com.skydrive.skydrive.service;


import org.springframework.stereotype.Service;

import com.skydrive.skydrive.dto.storage.StorageUsageResponse;
import com.skydrive.skydrive.entity.User;
import com.skydrive.skydrive.repository.FileRepository;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class StorageService {
    
    private final FileRepository fileRepository;
    private final UserService userService;


    public StorageUsageResponse getUsage(){

        User user = userService.getCurrentUser();

        long used = fileRepository.getUsedStorage(user.getId());

        long limit = user.getStorageLimit();

        long remaining = limit - used;

        double percentage = ((double) used / limit) * 100;

        return StorageUsageResponse.builder()
                .usedBytes(used)
                .limitBytes(limit)
                .remainingBytes(remaining)
                .usedPercentage(percentage)
                .build();
    }
}
